sources/cfsignal/src/system_check.rs

```diff
@@ -0,0 +1,64 @@
+use crate::error::{self, Result};
+use log::trace;
+use snafu::ResultExt;
+use std::process::Command;
+
+#[derive(Debug, Clone, Eq, PartialEq, Ord, PartialOrd)]
+pub(crate) struct SystemHealth {
+    /// Whether or not the system is healthy.
+    pub(crate) is_healthy: bool,
+    /// The system status description.
+    pub(crate) status: String,
+    /// The system status code.
+    pub(crate) exit_code: i32,
+}
```

Comment:
@webern do you have thoughts on making it easier to reuse this functionality from `metricdog`?

It seems suboptimal to have it almost but not quite the same - especially since the added abstraction here isn't supported by unit tests.

Reply:
> do you have thoughts on making it easier to reuse this functionality from metricdog?

I think the `SystemHealth` abstraction could probably be eliminated from this PR, or at least renamed in a way that makes it clear it's not the same thing as `SystemHealth` in `metricdog`.

We could have a library for querying systemd, I guess, but it seems like the shape of the queries are different enough that it would be an aggregating library rather than a code re-use library.

Reply:
but the name are not already different ?
metricdog use: `ServiceHealth`
cfsignal use: `SystemHealth`

or you are saying something different ?

Reply:
Sorry, I didn't realize the name was already different.